forch/faucet_event_client.py

```diff
@@ -151,22 +151,33 @@ def _debounce_port_event(self, event, port, active):
             self._handle_debounce(event, port, active)
             return
         state_key = '%s-%d' % (event['dp_id'], port)
-        if state_key in self._port_timers:
-            self._logger.debug('Port cancel %s', state_key)
-            self._port_timers[state_key].cancel()
-        if active:
-            self._handle_debounce(event, port, active)
-            return
-        self._logger.debug('Port timer %s = %s', state_key, active)
-        timer = threading.Timer(self._port_debounce_sec,
-                                lambda: self._handle_debounce(event, port, active))
-        timer.start()
-        self._port_timers[state_key] = timer
+        with self._buffer_lock:
+            if state_key in self._port_timers:
+                self._logger.debug('Port cancel %s', state_key)
+                self._port_timers[state_key].cancel()
+                if active:
+                    self._logger.info('Ignoring spurious port down event')
+                    self._port_timers.pop(state_key)
```

Comment:
This line is extra since the timer is also popped in handle_debounce. and shouldn't the line above say "port up" event?

Reply:
Or is it missing a return?